### PR TITLE
Decouple scda

### PR DESCRIPTION
**Pull request by gogonzo**
closes https://github.com/insightsengineering/teal.osprey/issues/213 

see also:
- https://github.com/insightsengineering/scda/pull/125
- https://github.com/insightsengineering/scda.2022/pull/123

### CODE REVIEW

**Review by lcd2yyz — COMMENTED** [review]
Overall comment: Not critical, but we can probably reduce the size of the example datasets? There were 50 unique subjects in the old example rASL, vs new rADSL has 400, which is feels a bit overkill.

File: data/rATE.rda
[the remark this answers is not in the document]
Comment (gogonzo) [reply data/rATE.rda:1]: What do you think about removing `"meddra"` also. It's not used in the examples anywhere.

File: R/g_swimlane.R [hunk 1]
```diff
@@ -30,16 +30,13 @@
...
-#'   dplyr::filter(PARAMCD == "OVRINV")
-#' ANL <- ASL %>% left_join(ARS, by = c("STUDYID", "USUBJID"))
+#' ARS <- filter(rADRS, PARAMCD == "OVRINV")
+#' ANL <- left_join(ASL, ARS, by = c("STUDYID", "USUBJID"), multiple = "all")
 #' anno_txt <- ASL[, c("ARMCD", "SEX")]
 #'
 #' g_swimlane(
 #'   bar_id = ASL$USUBJID,
-#'   bar_length = ASL$TRTDURD,
+#'   bar_length = as.integer(ASL$TRTEDTM - ASL$TRTSDTM),
```
[the remark this answers is not in the document]
Comment (gogonzo) [reply R/g_swimlane.R:39]: I needed to keep it `as.integer ` as g_waterfall fails when it is `difftime` class

**Review by lcd2yyz — APPROVED** [review]

File: data/rATE.rda
[the remark this answers is not in the document]
Comment (lcd2yyz) [reply data/rATE.rda:1]: Yes please drop if not used.

**Review by lcd2yyz — APPROVED** [review]
Can merge after last two request changes.

File: R/g_ae_sub.R [hunk 2]
```diff
@@ -31,12 +31,9 @@
...
 #'
 #' @examples
-#' library(scda)
 #' library(grid)
-#'
-#' cached_data <- synthetic_cdisc_data("latest")
-#' ADAE <- cached_data$adae
-#' ADSL <- cached_data$adsl
+#' ADAE <- rADAE
+#' ADSL <- rADAE
```
Comment (m7pr) [inline R/g_ae_sub.R:36]: typo fixed in here https://github.com/insightsengineering/osprey/pull/122

**Pull request merged**